src/main/java/com/cegeka/academy/service/challenge/ChallengeServiceImp.java

```diff
@@ -43,4 +51,25 @@ public void saveChallenge(ChallengeDTO challengeDTO) {
 
         logger.info("Challenge with id: " + challengeRepository.save(challenge).getId() + " has been saved");
     }
+
+    @Override
+    public Set<ChallengeDTO> getChallengesByUserId(long id) throws EmptyChallengeSetException {
+
+        Set<ChallengeDTO> userChallengesSet = new LinkedHashSet<>();
+
+        for(UserChallenge userChallenge : userChallengeRepository.findAllByUserId(id))
+        {
+            Challenge challenge = userChallenge.getChallenge();
+            userChallengesSet.add(ChallengeMapper.convertChallengeToChallengeDTO(challenge));
+        }
```

Comment:
De asemenea, poti folosi stream si map din Java 1.8  ca sa scapi de acest for. 

colectie.stream().map(ChallengeMapper.convertChallengeToChallengeDTO::challenge) ceva de genul acesta, dar poti citit pe internet despre stream si map in Java 1.8 ca sa faci conversia intre obiecte. Te rog sa scrii concluziile si pe grupul mare astfel incat sa-i informezi si pe ceilalti.